### DISCUSSION TITLE
Calling back into Idris from Java

### DISCUSSION THREAD

**Discussion by ska80** [mmhelloworld/idris-jvm#158] [General]
@mmhelloworld Thanks a lot again for your work on the JVM backend.

Is calling back into Idris from Java implemented?  This would enable integrating critical parts written in Idris in bigger Java applications (e.g., Spring Boot, etc.)

**Comment by mmhelloworld** [comment]
Thank you. Calling Idris from Java is not yet available. I recently came to know about `%export` directive and JS backend using it so I am planning to implement the same for JVM backend soon.

**Reply by ska80** [reply]
Thanks!

**Comment by mmhelloworld** [comment]
Calling Idris from Java is now available with version 0.7.0: [documentation](https://idris-jvm.readthedocs.io/en/latest/ffi/calling-idris-from-java.html). With that, we can now write a Spring Boot application completely in Idris! [Example](https://github.com/mmhelloworld/idris-spring-boot-example/blob/main/src/main/idris/Mmhelloworld/IdrisSpringBootExample/Main.idr).

**Reply by ska80** [reply]
@mmhelloworld Thank you very much for putting this much effort into implementing this feature. I am currently figuring out (tinkering with the `pom.xml` configuration ) how to add Idris parts to an existing Spring Boot application.
